firebase-authexchange/src/main/kotlin/com/google/firebase/authexchange/network/Routes.kt

```diff
@@ -0,0 +1,103 @@
+package com.google.firebase.authexchange.network
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Defines what an HTTP request may look like when interacting with the backend API for
+ * [FirebaseAuthExchange][com.google.firebase.authexchange.FirebaseAuthExchange].
+ */
+@Serializable internal sealed interface Request
+
+/** Defines what an HTTP response may look like when submitting a [Request]. */
+@Serializable internal sealed interface Response
+
+/**
+ * OIDC Credential.
+ *
+ * Useful for apps with external login already set-up, otherwise known as a headless OIDC flow.
+ *
+ * While this isn't the only OIDC flow that the API supports, it's the only flow that the mobile SDK
+ * supports- so there is no base class for credentials.
+ *
+ * @property idToken JWT encoded OIDC token returned from a third party provider
+ */
+@Serializable internal data class ImplicitCredentialsP(val idToken: String)
+
+/**
+ * Auth Token that can be used to access certain Firebase Services.
+ *
+ * This is merely the proto type for the network layer, and is completely separate from the user
+ * facing [AuthExchangeToken][com.google.firebase.authexchange.AuthExchangeToken].
+ *
+ * @property accessToken signed JWT containing claims that identify a user
+ * @property ttl the duration from the time this token is minted until its expiration
+ */
+@Serializable internal data class AuthExchangeTokenP(val accessToken: String, val ttl: String)
```

Comment:
What's the TTL format as a string?

Reply:
A long with an "s" appended to the end of it.

It's a `Duration` from the (our) protobuf package, defined [here](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/duration.proto).

Since this is translated directly from the JSON, I kept it as a string- as that's how protobuf exports it. You can read the header on `Duration` to learn more about that.

Technically I could write a custom deserializer for translating it to a `Duration` class, or something of the like, but I opted not to just cause that's how proto exports it, and all we do is translate it into a long for the application layer. Although I'm not against it. Thoughts?

Reply:
Thanks for the detailed explanation. No need to do any custom deserializer unless absolutely needed.